src/axom/sidre/docs/sphinx/query_data.rst

```diff
@@ -0,0 +1,98 @@
+caller can then inspect the information by retrieving values of specifically 
+named fields from the Node. This is often helpful for debugging.
+
+-------------
+Buffer Query
+-------------
+
+A datastore object can be queried for information about the buffer objects it
+contains. For example::
+
+  Datastore* ds = ...;
+
+  conduit::Node n;
+  ds->getBufferInfo(n);
+
+This method call inserts four fields into the Node that have numeric values
+accessible as type ``axom::IndexType``. For example::
+
+  using IndexType = axom::IndexType;
+
+  IndexType num_buffers = n["num_buffers"].value();
+  IndexType num_buffers_referenced = n["num_buffers_referenced"].value();
+  IndexType num_buffers_detached = n["num_buffers_detached"].value();
+  IndexType num_bytes_allocated = n["num_bytes_allocated"].value();
+
+The variables have the following values:
+
+  * ``num_buffers`` : Total number of buffer objects in the datastore.
+  * ``num_buffers_referenced`` : Number of buffers referenced by a view (i.e., attached to a view.
+  * ``num_buffers_detached`` : Number of buffers not referenced by a view. Note: ``num_buffers_detached`` = ``num_buffers`` - ``num_buffers_referenced``.
+  * ``num_bytes_allocated`` : Total number of bytes allocated in all buffers.
+
+--------------------
+Group Subtree Query
+--------------------
+
+A group can be queried for information about the data associated with it, or
+with the entire group subtree rooted at the group. For example::
+
+  Group* group = ...;
+
+  bool recursive = false; 
+
+  // get information about a single group
+  conduit::Node n;
+  group->getDataInfo(n, recursive);
+
+  // get information about entire subtree rooted at group
+  recursive = true;
+  conduit::Node n1;
+  group->getDataInfo(n1, recursive);
+
+Similar to the ``Datastore::getBufferInfo`` method described above, the 
+``Group::getDataInfo`` method inserts fields into the given Conduit Node
+that have numeric values accessible as type ``axom::IndexType``.  For example::
+
+  using IndexType = axom::IndexType;
+
+  IndexType num_groups = n["num_groups"].value();
+  IndexType num_views = n["num_views"].value();
+  IndexType num_views_empty = n["num_views_empty"].value();
+  IndexType num_views_buffer = n["num_views_buffer"].value();
+  IndexType num_views_external = n["num_views_external"].value();
+  IndexType num_views_scalar = n["num_views_scalar"].value();
+  IndexType num_views_string = n["num_views_string"].value();
+  IndexType num_bytes_assoc_with_views = n["num_bytes_assoc_with_views"].value();
+  IndexType num_bytes_external = n["num_bytes_external"].value();
+  IndexType num_bytes_in_buffers = n["num_bytes_in_buffers"].value();
+
+The variables have the following values describing the single group or
+entire group subtree:
+
+  * ``num_groups`` : Total number of groups.
+  * ``num_views`` : Total number of views.
+  * ``num_views_empty`` : Number of views with no associated buffer or data (may or may not be described).
```

Comment:
what does "described" mean in this context?

Reply:
Data description is a fundamental concept of Sidre Views and Buffers. The term is used throughout the source code and user documentation.

Reply:
It means that the view or buffer contains a description of the data it represents or contains, respectively. For example. 10 values of type double, etc.